libogc/network_wii.c

```diff
@@ -885,40 +885,56 @@ s32 net_send(s32 s, const void *data, s32 size, u32 flags)
 s32 net_sendto(s32 s, const void *data, s32 len, u32 flags, struct sockaddr *to, socklen_t tolen)
 {
 	s32 ret;
+	const s32 maxblocksize = (NET_HEAP_SIZE-64);
 	u8 * message_buf = NULL;
 	STACK_ALIGN(struct sendto_params,params,1,32);
 
 	if (net_ip_top_fd < 0) return -ENXIO;
 	if (tolen > 28) return -EOVERFLOW;
+	
+	for(ret = 0; ret < len;)
+	{
+		s32 send_size = len - ret;
+		if(send_size > maxblocksize)
+			send_size = maxblocksize;
+		
+		message_buf = net_malloc(send_size);
```

Comment:
move this allocation outside the loop.

Reply:
moved outside loop with max of either len or maxblocksize